File: ios/Classes/HovrNativePlatformViewPlugin.swift
import Flutter
import UIKit

public class HovrNativePlatformViewPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let factory = PhoneEntryViewFactory(messenger: registrar.messenger())
        registrar.register(
            factory,
            withId: PhoneEntryChannelConstants.viewType
        )
    }
}

File: android/src/main/kotlin/com/ridehovr/hovr_native_platform_view/HovrNativePlatformViewPlugin.kt
package com.ridehovr.hovr_native_platform_view

import io.flutter.embedding.engine.plugins.FlutterPlugin

class HovrNativePlatformViewPlugin : FlutterPlugin {
    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        binding.platformViewRegistry.registerViewFactory(
            PhoneEntryChannelConstants.VIEW_TYPE,
            PhoneEntryViewFactory(binding.binaryMessenger),
        )
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        // Platform views dispose individually.
    }
}
